Sources/OpenTelemetrySdk/Metrics/InstrumentBuilder.swift

```diff
@@ -40,8 +40,8 @@ public extension InstrumentBuilder {
     return self
   }
 
-  internal func swapBuilder<T: InstrumentBuilder>(_ builder: (inout MeterProviderSharedState, inout MeterSharedState, String, String, String) -> T) -> T {
```

Comment:
I believe if these are not `inout` then the internal `readerStorageRegisteries` and `callbackRegistration` will not be shared and updated in the MeterProvider

Reply:
But they are classes aren't they, the instances themselves are not changing, I thought it was a remaining if they were structs during development.

Reply:
yes, that may have been the issue.